### PR TITLE
Fixing links after check

### PR DESCRIPTION
**Pull request by moxiegirl**
- Ran a crawler over swarm docs to find broken links --- these are fixes

Signed-off-by: Mary Anthony <mary@docker.com>

### CODE REVIEW

**Comment by moxiegirl** [comment]
ping @vieux @aluzzardi @abronan Can y'all check this and merge if it ok with you?


author LGTM

**Comment by chanwit** [comment]
LGTM

File: docs/discovery.md [hunk 1]
```diff
@@ -208,7 +208,7 @@ And for the `nodes` discovery method.
 
 ## Contributing a new discovery backend
 
-You can contribute a new discovery backend to Swarm. For information on how to do this, see [our discovery README in the Docker Swarm repository](https://github.com/docker/swarm/blob/master/discovery/README.md).
+You can contribute a new discovery backend to Swarm. For information on how to do this, see <a href="https://github.com/docker/swarm/blob/master/discovery/README.md>our discovery README in the Docker Swarm repository</a>.
```
Comment (chanwit) [inline docs/discovery.md]: I tried and this link is not working.

File: docs/scheduler/strategy.md [hunk 2]
```diff
@@ -116,7 +116,10 @@ strategy prefers the node with most containers running.
 
 ## Docker Swarm documentation index
 
-- [User guide](https://docs.docker.com/swarm/)
-- [Discovery options](https://docs.docker.com/swarm/discovery/)
-- [Scheduler filters](https://docs.docker.com/swarm/scheduler/filter/)
-- [Swarm API](https://docs.docker.com/swarm/API/)
+
+- [User guide](/)
```
Comment (abronan) [inline docs/scheduler/strategy.md]: Tried and not working

File: docs/scheduler/strategy.md [hunk 3]
```diff
@@ -116,7 +116,10 @@ strategy prefers the node with most containers running.
...
+- [Discovery options](/discovery/)
+- [Scheduler strategies](/scheduler/strategy)
+- [Scheduler filters](/scheduler/filter)
+- [Swarm API](/api/swarm-api)
```
Comment (abronan) [inline docs/scheduler/strategy.md]: Same for this one

**Comment by moxiegirl** [comment]
@abronan are you building the docs?

**Comment by abronan** [comment]
Oh! nope was just checking quickly and didn't realize :smile: 

LGTM then

**Comment by chanwit** [comment]
Sorry for not understanding this. It means the links will be working fine after build, but not in the MD form on Github?

**Comment by abronan** [comment]
@chanwit when you build the doc it creates the right links and it works. Although I agree that this can be confusing and that I would assume that the links on the `.md` files in the repository should be working the same way :confused:

**Comment by chanwit** [comment]
@abronan @moxiegirl so should we fix the doc generator instead of hacking around `.md` files itself?

**Comment by thaJeztah** [comment]
> so should we fix the doc generator instead of hacking around .md files itself?

I think that should be the ultimate goal; not sure if we can realize that in the short term (releases), but I definitely think we should try to get that fixed.

@moxiegirl do we have a list of issues / examples with links that break (i.e., what Hugo created, and what it should be?)

**Comment by moxiegirl** [comment]
@chanwit Thank you for being concerned we really appreciate it. Our generator isn't broken.  @thaJeztah I know there have been comments about getting our Markdown to work as both GitHub repository pages and as a website. The pages in the `docs` directory aren't intended to as a website any more than the code you see there.  They have to be processed to become a website -- just like code is meant to built.

**Comment by chanwit** [comment]
I see @moxiegirl thank you for explanation in details.
Now I understand that `.md` files under `docs` are not necessary to links correctly in its MD form. The website version takes priority.

**Comment by aluzzardi** [comment]
Would it make sense **not** to name those `.md`? There's an implied understanding that `.md` files on a repo are supposed to be viewable on GitHub.

**Comment by moxiegirl** [comment]
@aluzzardi The files are viewable on GitHub.  What people are assuming is that because they can open a file in a browser and view its content *as if it were HTML content* that the page should act as a web page.  I've done a survey of projects similar to ours in size/or popularity.  Ruby on Rails puts a note on each page.

https://github.com/rails/rails/blob/master/guides/source/3_0_release_notes.md

Right now, I'm just opting for educating users.

**Comment by chanwit** [comment]
@moxiegirl This will solve the problem! It's a great idea to put a similar note on top of each page saying that users are expected to go to the rendered website.

**Comment by moxiegirl** [comment]
@chanwit Yes it would solve this issue.  Any solution tho can't be limited to this repository. @aluzzardi and the other Docker project owners need to be on board with it.  Andrea let us know what you think...

**Comment by chanwit** [comment]
LGTM

**Pull request merged**